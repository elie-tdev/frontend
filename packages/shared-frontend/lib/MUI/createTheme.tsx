import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
