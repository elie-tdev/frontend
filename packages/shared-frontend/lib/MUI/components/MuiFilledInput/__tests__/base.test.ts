import { MuiFilledInput } from '../'
import { createThemeTest } from '../../../createTheme'

describe('Checkbox', () => {
  const theme = createThemeTest()
  const component = MuiFilledInput(theme)

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
