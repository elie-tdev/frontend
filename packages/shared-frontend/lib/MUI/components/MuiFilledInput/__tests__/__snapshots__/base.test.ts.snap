// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should be consistent with previous versions:  1`] = `
Object {
  "defaultProps": Object {
    "disableUnderline": true,
  },
  "styleOverrides": Object {
    "focused": Object {
      "backgroundColor": "transparent",
      "borderWidth": 2,
    },
    "input": Object {
      "&.Mui-focused, &.Mui-focusVisible, &.Mui-active": Object {
        "backgroundColor": "transparent !important",
      },
      "backgroundColor": "transparent",
    },
    "root": Object {
      "&.Mui-error, &.MuiInputBase-colorSecondary.Mui-error": Object {
        "&.Mui-focused, &.Mui-active, &.Mui-focusVisible": Object {
          "boxShadow": "0px 0px 0px 2px #d32f2f",
        },
        "boxShadow": "0px 0px 0px 1px #d32f2f",
      },
      "&.Mui-focused, &.Mui-active, &.Mui-focusVisible": Object {
        "&.MuiInputBase-colorSecondary": Object {
          "boxShadow": "0px 0px 0px 2px #757575",
        },
        "boxShadow": "0px 0px 0px 2px #1B76D4",
      },
      "&.MuiInputBase-root": Object {
        "backgroundColor": "transparent",
      },
      "&:hover": Object {
        "boxShadow": "0px 0px 0px 1px rgba(0, 0, 0, 0.87)",
      },
      "backgroundColor": "transparent",
      "borderRadius": 8,
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxShadow": "0px 0px 0px 1px #bdbdbd",
    },
  },
}
`;
